pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -302,6 +322,16 @@ func (h *handler) patchStatus(obj runtime.Object, d data.Object, state rkev1.RKE
 		return nil, err
 	}
 
+	if state.JobComplete {
+		// Reset failureMessage and failureReason if they are not provided.
+		if _, ok := statusData["failureMessage"]; !ok {
+			statusData["failureMessage"] = ""
+		}
+		if _, ok := statusData["failureReason"]; !ok {
+			statusData["failureReason"] = ""
+		}
```

Comment:
Should we delete the key from the map instead?

Reply:
Yes. If they don't exist, then the loop below won't overwrite them and the failure information will stay on status. We want to reset failure reason/message to empty if the failure has been resolved.